[Chore] github actions 에서 종속성 캐싱 로직 추가

yonghyeun opened the pull request:
# 관련 이슈 번호
#35
# 설명
이슈 본문 , 댓글 참조 
# 첨부 파일 (Optional)

# 레퍼런스 (Optional)

aken-you committed:
[Refactor] IconButton에서 SelectedIconButton으로 네이밍 변경 (#46)

yonghyeun committed:
chore[#35] : github actions 에서 종속성 캐싱 로직 추가

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/b96081f2022219928bd9c23dff7a0c66/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/b96081f2022219928bd9c23dff7a0c66/shared-button.test.js
PASS browser: chromium ../../../../../tmp/b96081f2022219928bd9c23dff7a0c66/shared-input.test.js (5.504 s)

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        7.7 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

aken-you committed:
[Refactor] Button 컴포넌트 props에 fullWidth 추가 (#47)

yonghyeun committed:
chore[#35] : PR시 빌드 되지 않도록 깃허브 액션 수정

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/ec940d5c18043198f03f70fd985c41b4/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/ec940d5c18043198f03f70fd985c41b4/shared-input.test.js
PASS browser: chromium ../../../../../tmp/ec940d5c18043198f03f70fd985c41b4/shared-button.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        5.301 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

yonghyeun committed:
fix[#35] : cache 로직 변경 , id 추가하여 outputs 에 접근 가능하도록 변경

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/fe5adc6a93b958f8708068f50abc334c/shared-button.test.js
PASS browser: chromium ../../../../../tmp/fe5adc6a93b958f8708068f50abc334c/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/fe5adc6a93b958f8708068f50abc334c/shared-input.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        4.867 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

yonghyeun committed:
fix[#35] : cache 로직 변경 , name 추가

yonghyeun commented:
### ✅ Test-Storybook Results - Success
```
PASS browser: chromium ../../../../../tmp/d5774f3a959087e049d5b512bc99794b/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/d5774f3a959087e049d5b512bc99794b/shared-input.test.js
PASS browser: chromium ../../../../../tmp/d5774f3a959087e049d5b512bc99794b/shared-button.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        4.4 s
Ran all test suites.

```
🚀 [스토리북 확인하기](https://66bae48876a5167719ee1778-jqrdxsprac.chromatic.com)

yonghyeun committed:
[Feat] Input, TextArea label , statusText 로직 변경 (#45)

* fix[#41] : Input label, aria-label 분기문 추가

* feat[#41] : TextArea label  , arial 분기문 추가 , status Text 추가

* feat[#41] : Input with out label 스토리 추가

* feat[#41] : Input Focus 이벤트에만 statusText가 나타나도록 수정

* fix[#41] : onFocus , onBlur 이벤트 핸들러가 장착 되는 경우 고려

* feat[#41] : TextArea statusText 등장 로직 변경

* feat[#41] : TextArea label , essential props 추가

* fix[#41] : 스토리북 오타 수정

* feat[#41] : isFocused 상태 변경 로직 변경

* refactor[#41] : aria label null 병합 연산자 이용하여 리팩토링

* feat[#41] : Input , TextArea props 도 외부로 export

* feat[#41] : Input, TextArea fullWidth props 추가

* feat[#41] : Input , TextArea StatusText role ,aria-label 추가

* fix[#41] : TextArea autocomplete 추가

* fix[#41] : textArea type 수정 , fullwidth 컨디셔널 props로 변경

---------

Co-authored-by: kinsk2839 <96980857+aken-you@users.noreply.github.com>

aken-you commented on .github/workflows/storybook.yml:
```diff
@@ -14,17 +14,40 @@ jobs:
-      - uses: actions/setup-node@v4
+
+      - name: Cache npm dependencies
+        id: npm-cache
```
여기서 id는 어떤 역할을 하나요?!

aken-you commented on .github/workflows/storybook.yml:
```diff
@@ -14,17 +14,40 @@ jobs:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
npm 패키지를 캐시해서 chromatic 배포 시간을 줄이는 거군요 👍

yonghyeun replied:
호호호 배포 시간을 줄인다기보다 매 액션 시작 하기 전에 걸리는 불필요한 인스톨 시간을 줄이고자 합니다 

물론 이번 이슈에서 매번 `dev` 로 pr 이 날라 올 때 마다 배포하는 로직을 제거해서 불필요한 배포 시간도 줄이려 합니다.

`dev` 로 `push` 될 떄에만 배포하도록 하려구요

aken-you commented on .github/workflows/storybook.yml:
```diff
@@ -14,17 +14,40 @@ jobs:
+
+      - name: Check cache miss
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: echo 'npm cache missed!'
```
cache hit, cache miss에서 어떤 작업을 하는지 설명부탁드립니다!

yonghyeun replied:
해당 스텝은 단순히 캐시가 잘 작동하는지만 확인하려고 넣어봤어요 ㅎㅎ

깃허브 액션 안에선 캐시가 히트하면 위에서 설정한 키값의 패키지들을 이용하도록 하고

만약 캐시가 미스일 경우엔 새롭게 설치하고 캐시를 설정합니다.

aken-you reviewed (commented):
저도 이전 프로젝트에서 storybook 자동 배포 시간이 오래 걸렸는데, 이번 기회에 배워갑니다!

yonghyeun committed:
[Hotfix] Input,TextArea index.ts 파일 수정, EmailInput , PasswordInput 추가 (#51)

* hotfix[#50] : Input , TextArea index.ts index.ts 수정

* feat[#50] : EmailInput , PasswordInput 추가

* hotfix[#50] : visibility Icon 순서 변경

* feat[#50] : EmailInput control type 스토리북 추가

* fix#50] : Input,TextArea isError 일 경우엔 statusText 필수로 나타내기

* fix[#50] : Password Visibility control button tab index 추가

- 마우스 못쓰는 경우의 사람을 고려

* feat[#50] : PasswordInput storybook isError 추가

aken-you committed:
[Refactor] router 경로 상수 생성 (#57)

yonghyeun committed:
[Feat] useDebounce 훅 추가  (#58)

* feat[#55] : useDebounce 훅 추가

* fix[#55] : defaultValue => initialValue 이름 변경

* fix[#55] : useDebounce 주석 수정

yonghyeun committed:
chore[#56] : React Query 도입 (#59)

yonghyeun committed:
[Feat]  /login  경로 페이지 개발 (#61)

* chore[#52] : withRouter decorator 적용

- 모든 스토리에 react-router-dom 의 기능을 사용 가능하도록 데코레이터 추가
- 문서 : https://storybook.js.org/addons/storybook-addon-react-router-v6

* feat[#52] : login 페이지에 사용할 Icon 추가

* feat[#52] : BackwardNavigationBar 위젯 생성

* feat[#52] : 로그인 HyperLink 추가

* feat[#52] : /login 경로 페이지 생성

* fix[#52] : 발바닥 이모지 제거

* feat[#52] : BackwardNavigationBar onClick props 설정

* feat[#52] : 로그인 페이지 스토리북 추가

* fix[#52] : 로그인 하이퍼링크 entities -> features로 레이어 이동

* chore[#52] : withRouter 패키지 덜어내고 직접 구현하기

* feat[#52] : 세션 유지 정보를 담을 useAuthStore 생성

* feat[#52] : /login 경로 소셜 로그인 버튼 생성

- react query 를 이용하여 로그인 로직 생성
- TODO - 리다이렉션 로직 생성하기

* feat[#52] : /login 경로 로그인 페이지 생성

* chore[#52] : storybook-addon-remix-react-router 패키지 제거

- 해당 패키지를 제거하고 직접 구현했습니다.

* fix[#52] : 리뷰 사항 적용하여 변경

- 카멜케이스 유지하기 OAuth .. -> oAuth
- login-hyperlinks -> hyperlinks 파일명 변경
- 스토리북 tags 에서 pages 제거

* fix[#52] : OAuth 이름 롤백 oauth -> OAuth

aken-you committed:
[Feat] 이메일 회원가입 페이지 생성 및 이메일 검증 로직 구현 (#64)

* feat[#23] : 이메일 회원가입 등록 form 생성

* feat[#23] : 이메일 회원가입 등록 페이지 ui 생성

* feat[#23] : SignUpByEmailForm를 index.ts에서 내보내기

* feat[#23] : 이메일 회원가입 form에 관한 상태를 관리하는 커스텀 훅 생성

* refactor[#23] : visibility-off 수정

- mask 태그에는 width, height에 current 값이 없음

* feat[#23] : 이메일 인풋 테스트 코드 작성

* feat[#23] : 이메일 인풋 검증 로직 작성

- 이메일 형식에 맞지 않게 입력할 경우, statusText를 pink-500으로 표시

* refactor[#23] : 이메일 input 테스트 코드 수정

* fix[#23] : entities export 수정

yonghyeun committed:
[Hotfix] Input,TextArea stausText 높이 수정 ,BackwardNavigationBar props 수정 , storybook addon 수정 (#68)

* fix[#67] : Input , TextArea statusText 높이 정적으로 설정

* fix[#67] : BackwardNavigationBar props 수정

* fix[#67]:  storybook remix-react-router-dom addon 제거

* hotfix[#67] : Input,TextArea min-h-5 롤백

* fix[#67] : Input , TextArea statusText min-h-5로 롤백

aken-you committed:
[Refactor] 이메일 회원가입 form 검증 로직 구현 (#69)

* refactor[#23] : 이메일 회원가입 form에서 email과 password 검증 로직 수정

* test[#23] : email 테스트 코드 수정 및 password 테스트 코드 추가

* fix[#23] : 맨 첫 렌더링 때, 이메일 안내 문구가 보이는 문제 해결

* feat[#23] : 뒤로 가는 navigation bar 추가

* feat[#23] : 비밀번호 확인 input 검증 로직 구현

* refactor[#23] : email, password, passwordConfirm의 길이를 확인하는 변수 생성

* refactor[#23] : 각 input의 isError 조건문 수정

yonghyeun committed:
[Feat] /login/email 페이지 개발 (#65)

* chore[#52] : withRouter decorator 적용

- 모든 스토리에 react-router-dom 의 기능을 사용 가능하도록 데코레이터 추가
- 문서 : https://storybook.js.org/addons/storybook-addon-react-router-v6

* feat[#52] : BackwardNavigationBar 위젯 생성

* feat[#52] : 로그인 HyperLink 추가

* feat[#52] : /login 경로 페이지 생성

* fix[#52] : 발바닥 이모지 제거

* feat[#52] : BackwardNavigationBar onClick props 설정

* feat[#52] : 로그인 페이지 스토리북 추가

* fix[#52] : 로그인 하이퍼링크 entities -> features로 레이어 이동

* chore[#52] : withRouter 패키지 덜어내고 직접 구현하기

* feat[#52] : /login 경로 소셜 로그인 버튼 생성

- react query 를 이용하여 로그인 로직 생성
- TODO - 리다이렉션 로직 생성하기

* feat[#52] : /login 경로 로그인 페이지 생성

* chore[#52] : storybook-addon-remix-react-router 패키지 제거

- 해당 패키지를 제거하고 직접 구현했습니다.

* feat[#62] : LoginForm 생성

- Email 유효성 검증 로직 추가

* fix[#62] : 이전 커밋과 충돌 내역 병합

* fix[#62] : useDebounce 정규성 검사에서 사용하지 않기

* feat[#62] : LoginForm 컴포넌트 생성

* feat[#62] : /login/email 페이지 생성

* feat[#62] : 에러 핸들링 로직 추가

- + form 양식에서 persistLogin boolean 값 추가

* refactor[#62] : isError -> emailHasError , statusText -> emailStatusText 로 네이밍 변경

* feat[#62] : form 제출 전 유효성 검증을 통과하지 못할 경우 제출 막기

* [Hotfix] Input,TextArea stausText 높이 수정 ,BackwardNavigationBar props 수정 , storybook addon 수정 (#68)

* fix[#67] : Input , TextArea statusText 높이 정적으로 설정

* fix[#67] : BackwardNavigationBar props 수정

* fix[#67]:  storybook remix-react-router-dom addon 제거

* hotfix[#67] : Input,TextArea min-h-5 롤백

* fix[#67] : Input , TextArea statusText min-h-5로 롤백

* [Refactor] 이메일 회원가입 form 검증 로직 구현 (#69)

* refactor[#23] : 이메일 회원가입 form에서 email과 password 검증 로직 수정

* test[#23] : email 테스트 코드 수정 및 password 테스트 코드 추가

* fix[#23] : 맨 첫 렌더링 때, 이메일 안내 문구가 보이는 문제 해결

* feat[#23] : 뒤로 가는 navigation bar 추가

* feat[#23] : 비밀번호 확인 input 검증 로직 구현

* refactor[#23] : email, password, passwordConfirm의 길이를 확인하는 변수 생성

* refactor[#23] : 각 input의 isError 조건문 수정

* refactor[#62] : useLogin -> useEmailValidation 으로 리팩토링

* refactor[#62] : useLoginValidation -> useLoginForm 이름 변경 및 리팩토링

* refactor[#62] : useMutate 부분 네이밍 변경

* fix[#62] : useDebounce 롤백

* fix[#62] : useDebounce 다시 제거 , 유효성 검사 결과 상태가 아닌 객체로 관리

* fix[#62] : 리액트 쿼리 네임 컨벤션 지키기

* feat[#62] : 클라이언트 상태 로직 컴포넌트 내로 이동

---------

Co-authored-by: kinsk2839 <96980857+aken-you@users.noreply.github.com>

aken-you committed:
[Feat] Checkbox 컴포넌트 생성 (#73)

* feat[#72] : Checkbox에 사용할 아이콘 추가

* feat[#72] : Checkbox 컴포넌트 생성

* test[#72] : Checkbox 스토리북 작성

yonghyeun committed:
[Hotfix] 디자인 시스템에 맞춰 스타일 변경하기  (#77)

* fix[#76] : Input , TextArea label 색상 변경

* fix[#76] : global.css 에서 btn 폰트 시스템 변경

* fix[#76] : line-height 120% 수정

yonghyeun committed:
[Hotfix] mutationFn 에서 에러 처리 변경 (#78)

yonghyeun committed:
[Feat] /signup/pet-info 페이지 개발  (#74)

* feat[#71] : CloseNavigationBar 생성

* feat[#71] : 셀렉트 칩 공통 컴포넌트 개발

* feat[#71] : edit icon 추가

* feat[#71] : PetInfoForm 컴포넌트 생성

- zustand 를 이용해 Form 컴포넌트 렌더링 최적화

* feat[#71] : signup/pet-info 페이지 생성

* chore[#71] : 정적 파일 사용하기 위해 publicDir 추가

* feat[#71] : 프로필 배경 경로 window.location.origin 을 이용하도록 수정

* fix[#71] : shared/assets 에서 default-profile 제거

* feat[#71] : SelectChip 스토리북 추가

* fix[#71] : CloseNavigationBar props 타입 변경

* fix[#71] : useEffect 덜어내고 핸들링 하기

* feat[#71] : subscribeWithSelector 미들웨어 추가

* feat[#71] : pet-info 부분 유효성 검사 로직 추가

* fix[#71] : 스크린 리더기 사용자를 위한 visually-hidden 스타일 추가

* refactor[#71] : submit -> submitButton 으로 네이밍 변경

* refactor[#71] : setCharacterList 리팩토링

* fix[#71] : breed props 오타 수정 , isAlreadyExixted -> Selected 로 단어 변경

* fix[#71] : SelectChip props 구조 할당 방식으로 변경

* fix[#71] : edit svg 속성 current 로 변경

* fix[#71] : global.css visual only 삭제

* fix[#71] : 스토리북 docs 추가

* fix[#71] : HEX -> 디자인 시스템 색상으로 변경

* fix[#71] : else 사용 하징 않고 얼리리턴 사용

* fix[#71] : 이름이 입력되지 않았을 때 에러 처리가 되던 부분 수정

* fix[#71] : greed -> breed 오타 수정

* refactor[#71] : SelectChip 제어,비제어 컴포넌트로 확장

* fix[#71] : SelectChip 초기값 null 제거

- unControlled 이면서 상태 변경이 아닌 다른 onClick 메소드를 전달하는 경우 반영

* refactor[#71] : SelectChip 삼항 연산자 수정

* refactor[#71] : SelectChip Props 타입 수정

* fix[#71] : 스토리북 사용하지 않는 라이브러리 삭제

yonghyeun committed:
chore[#35] : 로컬 테스트 깃허브 액션 추가

yonghyeun committed:
fix[#35] : 로컬 테스트가 종료되면 포트를 닫는 액션 추가

yonghyeun committed:
fix[#35] : output 슬라이스 로직 변경

yonghyeun committed:
fix[#35] : 포트 열고 실행 될 때 까지 기다리는 wait 문 추가

yonghyeun committed:
chore[#35] : 로컬 테스트 전 6006 포트 닫는 액션 추가

yonghyeun committed:
chore[#70] : msw addon 추가 (#75)

yonghyeun committed:
fix[#35] : 백그라운드에서 스토리북을 실행하도록 & 기호 추가

yonghyeun committed:
chore[#35] : 공식문서 따라 npx http-server로 변경

yonghyeun commented:
### ✅ 현재 작업중인 브랜치의 스토리북 인터렉션 테스트 결과 - 모두 통과했습니다🔥
```
PASS browser: chromium ../../../../../tmp/49bd23b12ef529cfc2ade5794bb16e05/shared-input.test.js
PASS browser: chromium ../../../../../tmp/49bd23b12ef529cfc2ade5794bb16e05/shared-textarea.test.js
PASS browser: chromium ../../../../../tmp/49bd23b12ef529cfc2ade5794bb16e05/shared-button.test.js

Test Suites: 3 passed, 3 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        4.634 s
Ran all test suites.

```

yonghyeun committed:
fix[#35] : test-runner가 local 에서 작동하도록 기본 설정 변경

yonghyeun committed:
fix[#35] : storybook 시작 후 output 에 저장하는 단계 통합

- npm run test-storybook 에서 포트 6006 임을 명확히 명시

yonghyeun committed:
fix[#35] : start-storybook -> npx start-storybook 으로 변경

yonghyeun committed:
fix[#35] : 스토리북 실행 과정 병렬 -> 직렬적으로 변경

yonghyeun committed:
fix[#35] : 깃허브 액션에서 dev 모드로 스토리북 실행 시키기

yonghyeun commented:
### ❌ 현재 작업중인 브랜치의 스토리북 인터렉션 테스트 결과 - 통과하지 못한 테스트가 있어요!🫠
```
[33mThe CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
[33mThe CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
[33mThe CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
[33mThe CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
PASS browser: chromium src/shared/ui/navigationbar/NavigationBar.stories.tsx (6.282 s)
  ● Console

    console.log
      An error occurred in the following story, most likely because of a navigation: "shared/NavigationBar/Default". Retrying...

      at Object.<anonymous> (src/shared/ui/navigationbar/NavigationBar.stories.tsx:238:37)

PASS browser: chromium src/shared/ui/button/Button.stories.tsx (6.736 s)
  ● Console

    console.log
      An error occurred in the following story, most likely because of a navigation: "shared/Button/Default". Retrying...

      at Object.<anonymous> (src/shared/ui/button/Button.stories.tsx:238:37)

PASS browser: chromium src/shared/ui/textarea/TextArea.stories.tsx
PASS browser: chromium src/shared/ui/checkbox/Checkbox.stories.tsx
PASS browser: chromium src/shared/ui/input/Input.stories.tsx (8.584 s)
  ● Console

    console.log
      An error occurred in the following story, most likely because of a navigation: "shared/Input/Default". Retrying...

      at Object.<anonymous> (src/shared/ui/input/Input.stories.tsx:238:37)

PASS browser: chromium src/shared/ui/chip/SelectChip.stories.tsx
PASS browser: chromium src/features/auth/ui/LoginForm.stories.tsx
PASS browser: chromium src/features/auth/ui/SignUpByEmailForm.stories.tsx
PASS browser: chromium src/shared/ui/icon/Icon.stories.tsx
PASS browser: chromium src/shared/ui/badge/Badge.stories.tsx
PASS browser: chromium src/entities/auth/ui/Input.stories.tsx
PASS browser: chromium src/shared/ui/button/SelectedIconButton.stories.tsx
PASS browser: chromium src/widgets/navigationbar/ui/navigationbar.stories.tsx
PASS browser: chromium src/pages/login/login.stories.tsx
PASS browser: chromium src/features/auth/ui/hyperlinks.stories.tsx
PASS browser: chromium src/pages/login/email/email.stories.tsx
PASS browser: chromium src/pages/sign-up/pet-info/pet-info.stories.tsx

Test Suites: 17 passed, 17 total
Tests:       38 passed, 38 total
Snapshots:   0 total
Time:        17.957 s
Ran all test suites.

```

yonghyeun committed:
[Feat] 사용자의 라우트 경로를 추적하기 위한 HistoryTracker 컴포넌트 생성 (#79)

* fix[#60] : ROUTER_PATHS as const 로 타입 선언 방식 수정

* feat[#60] : useLoginProcessStore 추가

- 경로가 중복으로 쌓이는 것을 막기 위해 Set 자료구조를 사용했습니다.

* feat[#60] : /login 레이아웃 변경 , /login 페이지 loginProcess 과정 추가

* feat[#60] : HistoryTracker 컴포넌트 생성

- useLastNoneAuthStore 생성

* fix[#60] : 주석 오타 수정

* fix[#60] : 사용하지 않는 useLoginProcess 스토어 제거

* fix[#60] : 잘못 삭제된 AuthStore 롤백

* fix[#60] : 병합 충돌 예상 되는 부분 변경

* refactor[#60] : useRouteHistory 로 네이밍 변경

* refactor[#60] : useRouterHistoryStore 네이밍 변경

aken-you committed:
[Feat] 이메일 회원가입 api 로직 구현 (#81)

* feat[#23] : 이메일 회원가입 api 요청보내는 hook 생성

* feat[#23] : 이메일 회원가입 form에 submit handler 추가

* feat[#23] : 이메일로 회원가입 성공했을 경우, 기본정보 입력 페이지로 이동

* feat[#23] : auth store에 userId 추가

* feat[#23] : 이메일로 회원가입하면 auth store에 userId 저장

yonghyeun committed:
[Feat] /login 페이지 OAuth API 기능 추가 (#82)

* fix[#80] : authLogin Response 타입 변경 , AuthStore 변경

* feat[#80] : /login 페이지 OAuth 로그인 처리 기능 완료

* feat[#80] : 에러 핸들링 로직 추가

- TODO : 모달로 변경하기

* fix[#80] : 준회원1일 경우 개인정보 입력으로 이동 시키기

* fix[#80] : useAuthStore authorization -> token 으로 이름 변경

* fix[#80] : 서버 리스폰스 token -> authorization 으로 수정

* fix[#80] : LoginResponse 타입 변경

- useOAuthLogin -> useGetOAuthLogin 으로 네이밍 변경

aken-you committed:
[Feature] 기본 정보 입력 페이지 view 완성 (#83)

* feat[#66] : 약관 동의를 체크하는 Checkbox 컴포넌트 생성

* feat[#66] : 기본정보 입력 페이지에서 사용하는 아이콘 추가

* feat[#66] : 기본정보 입력 form 컴포넌트 view 완성

* feat[#66] : 기본정보 입력 페이지 view 구현

* feat[#66] : 옵션 리스트가 포함된 바텀 시트를 여는 컴포넌트 생성

* feat[#66] : 닉네임 Input statusText 설정

* feat[#66] : 닉네임 상태 추가

* test[#66] : 닉네임 input 테스트 코드 추가

aken-you committed:
[Feature] 기본정보 입력 api 로직 구현 (#87)

* refactor[#66] : 이메일 회원가입 api 요청보내는 hook 이동

- login.ts에서 signUp.ts로 이동

* feat[#66] : 기본정보 입력 api 요청보내는 로직 구현

yonghyeun committed:
[Refactor] LoginForm 컴포넌트 리팩토링 (#88)

* refactor[#86] : LoginForm 리팩토링

- useLoginFormStore 스토어와 함께 모듈화

* fix[#86] : nickname 도 useAuthStore 에 받도록 변경

* feat[#86] : API 가 정상적으로 응답받은 경우 스토어에 저장되는지 확인하는 테스트 코드 작성

* feat[#86] : API 요청이 실패한 경우의 테스트 코드 작성

* fix[#86] : 기본 statusText 부분 롤백

- 이전엔 "" 로 해두고 onFocus 시 초기화 하였는데 초기값으로 이메일 형식으로 입력해달라는 기본 값을 넣었습니다.

* fix[#86] : isNotValidEmail 명 수정

* fix[#86] : 테스트 코드 수정

* refactor[#86] : isNotValidEmail -> 긍정문인 isValidEmail 로 변경경

* fix[#86] : isValidEmail 부분 버그 나는 부분 수정

* fix[#86] : isValidEmail 파트 로직 변경

yonghyeun committed:
[Feat] /sign-up/pet-info 페이지 API 기능 추가  (#90)

* fix[#89] : PetInfoForm import 문 변경

* feat[#89] : 이모지인 경우 입력하지 못하게 하도록 기능 추가

* refactor[#89] : PetInfoForm export 경로 수정

* feat[#89] : usePetInfoForm ProfileImage 타입 url -> File 객체로 변경

* fix[#89] : SelectChip uncontrolled 일 때에도 onClick 이벤트 발생시키도록 수정

* feat[#89] : PetInfoForm 기본 기능 , 유효성 검사 , 상태 관리 테스트 코드 작성

* fix[#89] : usePostPetInfo 오타 수정

* feat[#89] : PetInfoForm API 요청 기능 추가

* fix[#89] : PetInfoForm.Profile 부분 File 객체를 받아 렌더링 하도록 수정

* fix[#89] : 제출 유효성 검사 로직 수정

* refactor[#89] : alertMessage 초기화 부분 변경

* feat[#89] : API 요청 테스트 코드 작성 완료

* feat[#89] : 입력 후 마지막에 접속한 경로로 라우팅하는 로직 추가

* fix[#89] : 엔드포인트 pet -> pets 수정

* refactor[#89] : mutationFn 콜백함수 내려주는 형태 변경

aken-you committed:
[Refactor] 기본 정보 입력 api 요청하는 로직 오타 수정 (#95)

* style[#66] : 기본 정보 추가 api 경로 오타 수정

* refactor[#66] : 기본 정보 추가 요청하는 함수에서 headers의 Authorization 수정

- 백엔드 api에 따라 토큰만 담아서 보내주는 걸로 수정

yonghyeun committed:
[Hotfix] OAuth 로그인 인증 과정 관련 로직 수정 (#94)

* fix[#93] : usePetInfoForm Authorization 헤더 추가

* fix[#93] : Oauth 로그인에 사용되던 useGetOAuthLogin 삭제

* feat[#93] : useOauthOnCookie 커스텀훅을 이용하여 OAuthLoginHyperLinks 로직 변경

* refactor[#93] : 불필요한 커스텀 훅으로 인한 상태 변경 로직 제거

- 변경 로직을 OAuthLoginHyperLinks 컴포 넌트 내에서만 처리 하도록 변경

* fix[#93] : 스토리북에서 token 로직 추가

* fix[#93] : Google OAuth 엔드포인트 오타 수정

* fix[#93] : AuthStore 타입 export 제거

yonghyeun committed:
[Feat] 페이지 라우팅 경로 수정 (#98)

* fix[#97] : 전역에 필요한 레이아웃들 main.tsx 에서 정의

* feat[#97] : router.ts 에서 레이아웃 경로 변경

* fix[#97] : /login 스토리북에서  LoginLayout 페이지 제거

* fix[#97] : BackwardButton 패딩값 large -> medium 으로 변경

* fix[#97] : /login/email 페이지 LoginLayout 제거

* fix[#97] : /login/email 페이지 LoginLayout 제거

* fix[#97] : router.ts 에서 path 상수화 이용하기

yonghyeun committed:
[Feat] / 페이지 레이아웃과 MainFooter 컴포넌트 생성 (#99)

* feat[#96] : GlobalNavigationBar 생성

* feat[#96] : Mobile , MainLayout 높이 뷰포트 설정

- 모바일 레이아웃의 경우 뷰포트 모두를 채우고 MainLayout 의 경우 상단 헤더와 하단 네비게이션 바를 제외한 메인 바가 모두 채우도록 설정

* feat[#96] : MainFooter 컴포넌트 생성

- `/ , map , community, my-page 라우트 경로 생성`
- MainFooter 컴포넌트 생성

* feat[#96] : Mobile , MainLayout 높이 뷰포트 설정

- 모바일 레이아웃의 경우 뷰포트 모두를 채우고 MainLayout 의 경우 상단 헤더와 하단 네비게이션 바를 제외한 메인 바가 모두 채우도록 설정

* fix[#96] : max-width 원상 복귀

* fix[#96] : 불필요한 html 문서 제거

* feat[#96] : / 경로 메인 섹션 스크롤 생성

yonghyeun committed:
chore[#35] : 크로매틱 빌드 후 빌드 된 url 코멘트 남기는 기능 추가
